src/filter/addons/filter-item-checkbox-helpers.ts

```diff
@@ -0,0 +1,137 @@
+import { FilterItemCheckboxOptionProps } from "../types";
+import { FlattenedFilterOption } from "./types";
+
+// =============================================================================
+// HELPER FUNCTIONS FOR NESTED OPTIONS
+// =============================================================================
+
+/**
+ * Builds a key path string from an array of keys
+ */
+export const buildKeyPath = (keyPath: string[]): string => {
+    return keyPath.join(".");
+};
+
+/**
+ * Flattens nested options into a flat array for rendering
+ */
+export const flattenNestedOptions = <T = FilterItemCheckboxOptionProps>(
+    nestedOptions: T[],
+    parentKeyPath: string[] = [],
+    valueExtractor?: (item: T) => string,
+    labelExtractor?: (item: T) => string
+): FlattenedFilterOption<T>[] => {
+    const flattened: FlattenedFilterOption<T>[] = [];
+
+    nestedOptions.forEach((option, index) => {
+        const value = valueExtractor
+            ? valueExtractor(option)
+            : (option as any).value;
+        const label = labelExtractor
+            ? labelExtractor(option)
+            : (option as any).label;
+        const options = (option as any).options;
+        const keyPath = [...parentKeyPath, String(value)];
```

Comment:
enforce typings as much as possible
```suggestion
        const value:string = valueExtractor
            ? valueExtractor(option)
            : (option as any).value;
        const label:string = labelExtractor
            ? labelExtractor(option)
            : (option as any).label;
        const options: any[] = (option as any).options;
        const keyPath = [...parentKeyPath, value];
```